src/stdlib_io_disp.fypp

```diff
@@ -0,0 +1,125 @@
+#:include "common.fypp"
+#:set RIL_KINDS_TYPES = REAL_KINDS_TYPES + INT_KINDS_TYPES + LOG_KINDS_TYPES
+
+submodule (stdlib_io) stdlib_io_disp
+
+    use, non_intrinsic :: stdlib_strings, only: format_string
+    implicit none
+    character(len=*), parameter :: fmt_r = '(*(g11.4, 1x))'
+    character(len=*), parameter :: fmt_c = '(*(g23.4, 1x))'
+    
+contains
+    
+    #:for kind, type in RIL_KINDS_TYPES
+    module procedure disp_0_${type[0]}$${kind}$
+        if(present(string)) print *, trim(string)
+        print fmt_r, val
```

Comment:
What about writing to a formatted unit using `write` instead of `print`? The unit could be passed optionally and defaults to `output_unit`.

Reply:
Originally, the main purpose of `disp` was to quickly print a simple low-dimensional array to our users **on the screen**, and it was simple and easy to use (user-friendly). Users can choose the redirection function of `bash` to switch the screen to other default output files, like:
```fortran
a.out <1.txt >2.txt
```
For outputting arrays to files, there are routines like `savetxt`, which are mainly for outputting two-dimensional arrays to ascii text files, and their functions are quite fixed. 
Yes, we can figure out how to keep `disp` simple and easy to use, and enable it to output files.

Reply:
Like @awvwgk I am also in favor of an `optional` argument `output_unit`